Update release lifecycle information (#305)
openedx/docs.openedx.org: source/community/release_notes/named_release_branches_and_tags.rst

Open edX Named Release Branches and Tags
########################################

> The Open edX community can share knowledge and improvements more easily when most people use the same stable, consistent version of the Open edX codebase. To that end, the Build Test Release working group creates "Open edX named releases", which have a release cycle on the order of six months between each release. The Open edX community, led by the Build Test Release working group, test the release for two months prior to it being made available to the community. The codebase is also extensively tested by the 2U/edX team, who deploy the code to the edx.org website multiple times a day.

Reviewer: ```suggestion
The Open edX community can share knowledge and improvements more easily when most people use the same stable, consistent version of the Open edX codebase. To that end, the Build-Test-Release working group creates an "Open edX named release" every six months. The Open edX community, led by the Build Test Release working group, tests the release for two months prior to it being made available to the community. The codebase is also extensively tested by the 2U/edX team, who deploys the code to the edx.org website multiple times a day.
```
